tools/chpl-language-server/src/chpl-language-server.py

```diff
@@ -863,6 +979,20 @@ def get_use_or_def_segment_at_position(
 
         return None
 
+    def scope_at_position(
+        self, position: Position
+    ) -> Optional[ScopedNodeAndRange]:
+        """
+        Given a position, return the scope that contains it.
+        """
+        found = None
+        for s in self.scope_segments.elts:
+            if s.rng.start <= position <= s.rng.end:
+                found = s
+            if s.rng.start > position:
+                break
+        return found
+
```

Comment:
I think this points to an issue with our segment list implementation (it should be a binary search there too). For now this is good, but I am looking into it.